File: robot_examples/go/rest_api/swagger/EventScan.go
package swagger

import (
)

type EventScan struct {
    EventType  Number  `json:"eventType,omitempty"`
    ActivationTime  float32  `json:"activationTime,omitempty"`
    Direction  float32  `json:"direction,omitempty"`
    Semiaperture  float32  `json:"semiaperture,omitempty"`
    ScanMaxDistance  float32  `json:"scanMaxDistance,omitempty"`
    Robot  RobotInfo  `json:"robot,omitempty"`
    HitRobot  RobotInfo  `json:"hitRobot,omitempty"`
    
}

File: robot_examples/haskell-servant/rest_api/lib/Model/EventScan.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.EventScan
    ( EventScan (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.Event
import Model.Number
import Model.RobotInfo


data EventScan = EventScan
    { eventType :: Number
    , activationTime :: Float
    , direction :: Float
    , semiaperture :: Float
    , scanMaxDistance :: Float
    , robot :: RobotInfo
    , hitRobot :: RobotInfo
    } deriving (Show, Eq, Generic)

instance FromJSON EventScan
instance ToJSON EventScan
instance Arbitrary EventScan where
    arbitrary = EventScan <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
